jobbergate-slurp/slurp/migrators/job_scripts.py

```diff
@@ -2,61 +2,119 @@
                 job_script_description,
                 job_script_owner_email,
                 application_id,
                 created_at,
                 updated_at
             )
-            values (
-                %(name)s,
-                %(description)s,
-                %(owner_email)s,
-                %(application_id)s,
-                %(created)s,
-                %(updated)s
-            )
-            returning id
-            """,
-            dict(
-                name=job_script["job_script_name"],
-                description=job_script["job_script_description"],
-                owner_email=owner_email,
-                application_id=nextgen_application_id,
-                created=job_script["created_at"],
-                updated=job_script["updated_at"],
+            values {}
+            """.format(
+                mogrified_params
             ),
         )
 
-        nextgen_jobscript_id = result.fetchone()["id"]
-        job_scripts_map[job_script["id"]] = nextgen_jobscript_id
+    logger.success("Finished migrating job_scripts")
+
+
+async def transfer_job_script_files(legacy_job_scripts):
+    """
+    Transfer job-script files from a column at the legacy database to nextgen s3.
+    """
+    logger.info("Start migrating job-script files to s3")
+
+    main_filename = "application.sh"
+    s3_key_template = f"job-scripts/{{job_script_id}}/main-file/{main_filename}"
 
-        s3man.nextgen[nextgen_jobscript_id] = job_script["job_script_data_as_string"]
+    async def transfer_helper(bucket, job_script_data_as_string, job_script_id):
+        """
+        Helper function that handles the transfer of a single job-script.
+        """
+        with handle_errors(
+            f"Error getting the job-script content from the JSON ({job_script_id=}): {job_script_data_as_string}"
+        ):
+            unpacked_data = json.loads(job_script_data_as_string)
+            job_script_content = unpacked_data[main_filename]
 
-    logger.success("Finished migrating job_scripts")
-    return job_scripts_map
+        require_condition(bool(job_script_content), f"Empty job script content for {job_script_id=}")
+
+        s3_key = s3_key_template.format(job_script_id=job_script_id)
+
+        with handle_errors(f"Error uploading {job_script_id=} content to {s3_key=}"):
+            await bucket.upload_fileobj(BytesIO(job_script_content.encode("utf-8")), s3_key)
+
+        return job_script_id
+
+    async with s3_bucket(is_legacy=False) as bucket:
+
+        tasks = (
+            asyncio.create_task(
+                transfer_helper(bucket, job_script["job_script_data_as_string"], job_script["id"])
+            )
+            for job_script in legacy_job_scripts
+        )
+
+        results = await asyncio.gather(*tasks, return_exceptions=True)
+
+    for r in results:
+        if isinstance(r, Exception):
+            logger.warning(str(r))
+
+    transferred_ids = {i for i in results if isinstance(i, int)}
```

Comment:
you can implement something similar as my suggestion to avoid looping over unnecessary fields

Reply:
Another filter would be another implicit loop. But I see the point, I refactored to code to get both exceptions and transferred ids in just one loop. In addition, it can catch any unexpected result, which is nice.